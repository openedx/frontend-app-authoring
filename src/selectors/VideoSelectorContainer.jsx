import React from 'react';
import PropTypes from 'prop-types';
import { useParams } from 'react-router-dom';
import { VideoSelectorPage } from '@edx/frontend-lib-content-components';
import { getConfig } from '@edx/frontend-platform';

const VideoSelectorContainer = ({
  courseId,
}) => {
  const { blockId } = useParams();
  return (
    <div className="selector-page">
      <VideoSelectorPage
        blockId={blockId}
        courseId={courseId}
        studioEndpointUrl={getConfig().STUDIO_BASE_URL}
        lmsEndpointUrl={getConfig().LMS_BASE_URL}
      />
    </div>
  );
};

VideoSelectorContainer.propTypes = {
  courseId: PropTypes.string.isRequired,
};

export default VideoSelectorContainer;
