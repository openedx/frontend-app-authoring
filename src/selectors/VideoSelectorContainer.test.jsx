import React from 'react';
import { shallow } from 'enzyme';
import VideoSelectorContainer from './VideoSelectorContainer';

jest.mock('@edx/frontend-lib-content-components', () => ({ VideoSelectorPage: () => 'HeaderTitle' }));

jest.mock('react-router', () => ({
  ...jest.requireActual('react-router'), // use actual for all non-hook parts
  useParams: () => ({
    blockId: 'company-id1',
    blockType: 'html',
  }),
}));

const props = { courseId: 'cOuRsEId' };

describe('Video Selector Container', () => {
  describe('snapshots', () => {
    test('rendering correctly with expected Input', () => {
      expect(shallow(<VideoSelectorContainer {...props} />)).toMatchSnapshot();
    });
  });
});
