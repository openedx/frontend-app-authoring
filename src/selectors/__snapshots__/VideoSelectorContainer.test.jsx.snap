// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Video Selector Container snapshots rendering correctly with expected Input 1`] = `
<div
  className="selector-page"
>
  <VideoSelectorPage
    blockId="company-id1"
    courseId="cOuRsEId"
    lmsEndpointUrl="http://localhost:18000"
    studioEndpointUrl="http://localhost:18010"
  />
</div>
`;
